### PR TITLE
WorkloadSpread support customized workload

### PR DESCRIPTION
**Pull request by veophi**
### Ⅰ. Describe what this PR does
**Require `scale sub-resource` support for customized workload**

### CODE REVIEW

File: pkg/util/controllerfinder/pods_finder.go [hunk 1]
```diff
@@ -102,8 +99,34 @@ func (r *ControllerFinder) GetPodsForRef(apiVersion, kind, ns, name string, acti
...
+			FieldSelector: fields.SelectorFromSet(fields.Set{fieldindex.IndexNameForOwnerRefUID: string(uid)}),
+		}
+		pods, err := listPods(&listOption)
+		if err != nil {
+			return nil, -1, err
+		}
+		matchedPods = append(matchedPods, pods...)
 	}
 
+	if labelSelector != nil && len(matchedPods) == 0 { // try again using label selector for scale sub-resource
```
Comment (furykerry) [inline pkg/util/controllerfinder/pods_finder.go:119]: why retry using the pod label selector ? plz comment about the cases

File: pkg/util/configuration/types.go [hunk 2]
```diff
@@ -75,3 +76,13 @@ func (p *PPSWatchWatchCustomWorkloadWhiteList) IsDefaultSupport(apiVersion, kind
...
 }
+
+type WSCustomWorkloadWhiteList struct {
+	Workloads []CustomWorkload `json:"workloads,omitempty"`
+}
+
+type CustomWorkload struct {
+	schema.GroupVersionKind `json:",inline"`
+	SubResources            []schema.GroupVersionKind `json:"subResources,omitempty"`
+	ReplicasPath            string                    `json:"replicasPath,omitempty"`
```
[the remark this answers is not in the document]
Comment (veophi) [reply pkg/util/configuration/types.go]: done

File: pkg/util/controllerfinder/pods_finder.go (hunk 1 above)
Comment (veophi) [reply pkg/util/controllerfinder/pods_finder.go]: done

**Review by furykerry — COMMENTED** [review]
/lgtm

**Pull request merged**